src/openrct2/interface/Viewport.cpp

```diff
@@ -1639,35 +1639,36 @@ InteractionInfo set_interaction_info_from_paint_session(paint_session* session,
  * viewport: edi
  */
 void get_map_coordinates_from_pos(
-    ScreenCoordsXY screenCoords, int32_t flags, CoordsXY& mapCoords, int32_t* interactionType, TileElement** tileElement,
+    const ScreenCoordsXY& screenCoords, int32_t flags, CoordsXY& mapCoords, int32_t* interactionType, TileElement** tileElement,
     rct_viewport** viewport)
 {
     rct_window* window = window_find_from_point(screenCoords);
     get_map_coordinates_from_pos_window(window, screenCoords, flags, mapCoords, interactionType, tileElement, viewport);
 }
 
 void get_map_coordinates_from_pos_window(
-    rct_window* window, ScreenCoordsXY screenCoords, int32_t flags, CoordsXY& mapCoords, int32_t* interactionType,
+    rct_window* window, const ScreenCoordsXY& screenCoords, int32_t flags, CoordsXY& mapCoords, int32_t* interactionType,
     TileElement** tileElement, rct_viewport** viewport)
 {
     InteractionInfo info{};
+    auto dynamicScreenCoords = screenCoords;
     if (window != nullptr && window->viewport != nullptr)
     {
         rct_viewport* myviewport = window->viewport;
-        screenCoords.x -= (int32_t)myviewport->x;
-        screenCoords.y -= (int32_t)myviewport->y;
-        if (screenCoords.x >= 0 && screenCoords.x < (int32_t)myviewport->width && screenCoords.y >= 0
-            && screenCoords.y < (int32_t)myviewport->height)
+        dynamicScreenCoords.x -= (int32_t)myviewport->x;
```

Comment:
This function seems especially pointless to use const, considering how heavily it modifies the coords.